platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/merkle/VirtualRootNode.java

```diff
@@ -840,6 +847,7 @@ public void put(final K key, final V value) {
         final VirtualLeafRecord<K, V> leaf = new VirtualLeafRecord<>(path, key, value);
         cache.putLeaf(leaf);
         statistics.countUpdatedEntities();
+        assert currentModifyingThreadRef.compareAndSet(Thread.currentThread(), null);
```

Comment:
I assume all these calls to reset `currentMofdifyingThreadRef` to null should be done in try-finally (e.g. there is a `return null` call in the middle of `remove()`, which leaves the ref not null after the method is finished).

Reply:
Added try/finally blocks for all asserts